src/setupTests.ts

```diff
@@ -2,3 +2,14 @@ import Enzyme from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
 
 Enzyme.configure({ adapter: new Adapter() });
+
+const globalAny: any = global;
+class MockEncoder {
+  encode() {
+    return jest.fn();
+  }
+}
+const mockCrypto = { subtle: { digest: () => jest.fn() } };
+
+globalAny.window.TextEncoder = MockEncoder;
```

Comment:
How much of the tests require these mocks? If it only affects the robot tests its better to set them up in their respective test cases.

Reply:
It affects any test with colorManager instance involved. Currently it only affects robotTrajectoryOverlay and color tests but I will leave it here for now as we might need to access colorManager in other test in the future when improving code coverage or testing components that require colorManager.